### PR TITLE
feat(Amount): Amount component migration

### PR DESCRIPTION
**Pull request by saurabhdaware**

### CODE REVIEW

**Comment by chaitanyadeorukhkar** [comment]
@saurabhdaware missing space in story
Screenshot 2023-12-28 at 3 08 08 PM

**Comment by chaitanyadeorukhkar** [comment]
Lets keep some visual gap between each `type`? 
Screenshot 2023-12-28 at 3 08 43 PM

File: packages/blade/src/components/Amount/amountTokens.ts [hunk 1]
```diff
@@ -1,48 +1,102 @@
...
+
+type AmountBodyProps = {
+  type?: 'body';
+  size?: Extract<AmountSizes, 'xsmall' | 'small' | 'medium' | 'large'>;
+  weight?: Extract<BaseTextProps['fontWeight'], 'regular' | 'medium' | 'semibold'>;
+};
+
+type AmountTypeProps = AmountDisplayProps | AmountHeadingProps | AmountBodyProps;
+
+const normalAmountSizes: Record<
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Amount/amountTokens.ts:26]: we just need to add run time error for type values that don't have corresponding size values

**Comment by kamleshchandnani** [comment]
@saurabhdaware can you rebase and resolve conflicts?

**Pull request merged**